File: customer-system-backend/src/main/java/com/assignment/customer_system_backend/service/AdminService.java
package com.assignment.customer_system_backend.service;

import com.assignment.customer_system_backend.entity.AdminEntity;
import com.assignment.customer_system_backend.entity.CustomerEntity;
import com.assignment.customer_system_backend.model.Admin;
import com.assignment.customer_system_backend.repository.AdminRepo;
import org.springframework.beans.BeanUtils;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class AdminService {
    @Autowired
    AdminRepo adminRepo ;

    public Admin signUp(Admin admin) {
        AdminEntity adminEntity = new AdminEntity();
        BeanUtils.copyProperties(admin,adminEntity);
        adminRepo.save(adminEntity);
        return admin;
    }

    public boolean login(String emailId, String password) {
        if (adminRepo.existsById(emailId)) {
            AdminEntity adminEntity = adminRepo.findById(emailId).get();
            return password.equals(adminEntity.getPassword());
        }
        return false;
    }
}

File: customer-system-frontend/src/services/AdminService.js
import axios from "axios";

const ADMIN_API_BASE_URL = "http://localhost:8080/admin";
class AdminService {
  signUp(admin) {
    return axios.post(`${ADMIN_API_BASE_URL}/signup`, admin);
  }

  login(admin) {
    return axios.post(`${ADMIN_API_BASE_URL}/login`, admin);
  }
}

export default new AdminService();
